Assets/Scripts/WeaponSpawner.cs

```diff
@@ -24,20 +29,63 @@ void Start()
         {
             mHasBeenCollected = false;
             mRenderer = GetComponent<SpriteRenderer>();
-            mCollider = GetComponent<BoxCollider2D>();
             mPhotonView = GetComponent<PhotonView>();
+
+            mCollider = GetComponent<BoxCollider2D>();
+
+            float halfX = mCollider.bounds.extents.x;
+            float halfY = mCollider.bounds.extents.y;
+            float offsetX = mCollider.offset.x;
+            float offsetY = mCollider.offset.y;
+
+            mTLCorner = transform.TransformPoint(offsetX - halfX, offsetY + halfY, 0f);
+            mBRCorner = transform.TransformPoint(offsetX + halfX, offsetY - halfY, 0f);
         }
 
-        void OnTriggerEnter2D(Collider2D other)
+        void Update()
         {
-            if (other.tag == Tags.PLAYER &&
-                other.gameObject.GetComponent<WeaponInventory>().CanEquip())
+            if (InputWrapper.Instance.EquipWeaponPressed)
+            {
+                CheckForPlayerCollisions();
+            }
+        }
+
+        void CheckForPlayerCollisions()
+        {
+            Collider2D[] playersInBoundingBox = Physics2D.OverlapAreaAll(mTLCorner, mBRCorner, mPlayerLayer);
+            Collider2D player = FindPlayerWhoPressedEquip(playersInBoundingBox);
+            if (player != null)
             {
                 Debug.Log("weapon");
                 // this is an id that maps 1-to-1 with the player who collected the weapon
-                int viewId = other.gameObject.GetComponent<PhotonView>().viewID;
-                mPhotonView.RPC("OnWeaponTriggered", PhotonTargets.MasterClient, viewId);
+                int playerViewId = player.gameObject.GetComponent<PhotonView>().viewID;
+                mPhotonView.RPC("OnWeaponTriggered", PhotonTargets.MasterClient, playerViewId);
+                // disable swap prompt for each player currently in the weapon's bounding box
+                for (int i = 0; i < playersInBoundingBox.Length; i++)
+                {
+                    int viewId = playersInBoundingBox[i].gameObject.GetPhotonView().viewID;
+                    mPhotonView.RPC("DisableSwapPrompt", PhotonTargets.All, viewId);
+                }
+            }
+        }  
+        
+        Collider2D FindPlayerWhoPressedEquip(Collider2D[] playersInBoundingBox)

```

Comment:
I think this method name is a little confusing. Maybe we could rename it more closely to what its actually doing? Something like FilterForLocalPlayer since the filter here isn't about PressingEquip but just about being local.